### ISSUE TITLE
Check for custom extended key usage

### ISSUE THREAD

**Issue by neusdan** [randombit/botan#873]
I'm porting our application to Botan 2.0.1 and have a problem with the extended key usage check.

Before #706 it was possible to add OIDs at runtime. My application code looked like this:

```c++
Botan::OIDS::add_oid( Botan::OID( MY_OID ), MY_DESCRIPTION );
cert.has_ex_constraint( MY_DESCRIPTION );
```

Now the OID lookup fails in the `has_ex_constraint()` method.

@randombit I can remember that you already proposed a solution to this, but unfortunately i can't remember and don't find it anymore on GitHub.

**Comment by randombit** [comment]
I can't remember what I suggested. :(

Can you call `cert.ex_constraints()` and just do the search within your application?

Looking at the header, X509_Certificate API is kind of strange and restricting ATM. Working on a patch to improve things somewhat for 2.1.

**Comment by neusdan** [comment]
> Can you call cert.ex_constraints() and just do the search within your application?

No :-( 

`X509_Certificate::ex_constraints()`performs a lookup and returns only those OIDs Botan knows about.

**Comment by randombit** [comment]
WIP in `x509-member-data`, will post a PR when ready

**Comment by neusdan** [comment]
Thanks a lot!

**Comment by randombit** [comment]
Fixed now in 2.4.0, adds ``const std::vector<OID>& X509_Certificateextended_key_usage()``

Also in 2.5.0 you can add OIDs again ;)
